cmd/dagger/cmd/do.go

```diff
@@ -85,13 +85,9 @@ var doCmd = &cobra.Command{
 		}
 
 		actionFlags := getActionFlags(action)
-
+		actionFlags.Parse(args)
```

Comment:
What's the diff between what we tried and now? We are still parsing flags two time

Reply:
This works because we're only parsing the `actionFlags` FlagSet that *only* handles the action flags. Before we were calling `cmd.Flags().Parse()` which was re-parsing all the arguments twice and therefore duplicating the `--cache-to` value by triggering https://github.com/spf13/pflag/issues/244. Does it make sense?

Reply:
Sure! Thank you :D